### ISSUE TITLE
Disabling "stale bots" (prow lifecycle plugin) on Tektoncd organization

### ISSUE THREAD

**Issue by vdemeester** [tektoncd/community#1092]
TL;DR I am proposing to **disable the `lifecycle` plugin, that "Closes, reopen, flag and/or unflag an issue or PR as frozen/stale/rotten" from most if not all `tektoncd` repositories**, as I feel it's more of a problem than a solution.

The `lifecycle` plugin from prow is essentially what is called elsewhere a "stale bot". [Probot stale](https://github.com/probot/stale) was an example, on GitHub, there is an *official* action for this [`actions/stale`](https://github.com/actions/stale). This plugin (and a "stale bot") in general, is set in place to automatically triage issues and pull-request from a repository. It comes from good intention, it is supposed to help maintainers manage the amount of issues and pull-request from their repository.

There is a [bunch](https://drewdevault.com/2021/10/26/stalebot.html) [of](https://fvsch.com/stale-bots) [articles](https://www.tzeyiing.com/posts/stale-bot-for-projects/) [that](https://www.theozimmermann.net/2021/11/known-issues/) talk about this, but I'll try to summarize here why I think they are counter-productive for us (tektoncd community).

- **From the reporter perspective**
	- When looking for an existing bug report (or issue), one might find a closed issue, and think it was already implemented or already fixed.
	- When finding a similar issue that the reporter's issue, which has been closed by the bot/lifecycle plugin, there is no reason why it was closed.
		- Was it fixed ?
		- Was in closed as "not fixeable" ?
	- It may create some confusion when reporting an issue that is already close : should I create a new issue or should I comment on the existing one ?
- **From the maintainer perspective**
	- It tends to get in the way:
		- There can be a lot of "non-useful" comments on a valid issue (that is old), like `/remove-lifecycle …`, …
	- It doesn't help that much. For example, on https://github.com/tektoncd/pipeline/issues/5776, the issue became stale and rotten. Then new activities happens on the issue, but we forget to make sure the lifecycle is resetted. And thus, soon after, while being activiley discussed, it gets automatically closed (8 comments out of 20 are related to "lifecycle" management as well).
	- It potentially adds valid issues and feature request that didn't get enough traction for a "relatively" long time.
	- It reduces maintainers important triaging work.
		- We almost never close issue ourselves (even though we think they should be closed for a valid reason).
		- We don't have a "process" to qualify issues.
		- We don't really use lot's of labels we have to actually help other maintainers and contributors picking work (things like `priority/*`, `help wanted`, …).
- And more…
	- As discussed in different articles, it gives a "false" sense of an active project where issues get closed on a relatively _fast_ pace.

(Sidenote: there is also the GitHub Discussions that are not going through any triaging process or anything, but that's a topic for another day)

**Comment by jerop** [comment]
Agreed. We added the stale bot to help with surfacing issues that have not had activity in a while so that they can be addressed or updated, but it seems that it's not increased contributor engagement on the issues. Makes sense to remove it. I wonder if we can address the issue in another way? Maybe start a triaging rotation?

**Comment by vdemeester** [comment]
Let me ping all maintainers to see if there is any concern 👼🏼 
cc @tektoncd/core-maintainers @tektoncd/chains-maintainers @tektoncd/triggers-maintainers @tektoncd/results-maintainers @tektoncd/cli-maintainers @tektoncd/operator-maintainers @tektoncd/hub-maintainers @tektoncd/catalog-maintainers

**Comment by vdemeester** [comment]
Please raise any concerns about the above proposals. If there is none in a week (so ~7th of November), I'll go ahead and de-activate the prow plugin.

**Comment by AlanGreene** [comment]
fyi @tektoncd/dashboard-maintainers

**Comment by vdemeester** [comment]
(oh sorry @AlanGreene, I forgot about dashboard 🙀 🙏🏼 )
